libraries/tlv-account-resolution/src/seeds.rs

```diff
@@ -0,0 +1,320 @@
+//! Types for managing seed configurations in TLV Account Resolution
+//!
+//! The largest possible seed configuration is 3 bytes (`InstructionArg`).
+//! This means that an `AccountMetaPda` can store up to 10 seed configurations.
+
+use {crate::error::AccountResolutionError, solana_program::program_error::ProgramError};
+
+/// Enum to describe a required seed for a Program-Derived Address
+#[derive(Clone, Debug, PartialEq)]
+pub enum Seed {
+    /// Uninitialized configuration byte space
+    Uninitialized,
+    /// A literal hard-coded argument
+    Literal {
+        /// The literal value repesented as a vector of bytes.
+        /// For example, if a literal value is a string literal,
+        /// such as "my-seed", this value would be
+        /// `"my-seed".as_bytes().to_vec()`.
+        bytes: Vec<u8>,
```

Comment:
I can totally add this, but since it's being packed without you really knowing, I didn't think it made sense to include in the docs/comments on the enum itself. My thinking was kinda: if you're looking at how to use this enum, it doesn't have length in it. wdyt?

Reply:
Maybe it's just me, but when I'm planning out my seeds, I'll look at the `Seed` type and start counting up everything, but then I see `Vec` and think, "how's that even encoded?" I'd put in just a sentence about how it's encoded